### PR TITLE
fix(nextly): store number fields as exact decimals when requested

### PR DESCRIPTION
**Pull request by mobeenabdullah**
## Summary

Code-first number fields could only store integers: the field/column mapping sent every code-first `number` to an INTEGER column unless the value carried the builder-only `options.format === "float"` flag. So a code-first price field silently truncated `19.99` to `19`, even though its JSDoc advertised prices and `step: 0.01`.

This is finding **P0-G** from the fields/schema audit, routed to (but not addressed by) the drizzle-v1 migration. See `tasks/drizzle-v1-findings-verification.md` in the planning repo for the full verification.

## What changed

- New opt-in `dbType: "decimal"` on `NumberFieldConfig`, with optional `precision`/`scale` (default `DECIMAL(10, 2)`). Integer stays the default, so **existing fields are unchanged** (no migration).
- New `decimal` column kind in `field-column-descriptor.ts`, rendered as exact `numeric(p,s)` on Postgres, `decimal(p,s)` on MySQL, and `numeric` (NUMERIC affinity) on SQLite, read back as a JS number to match the number-field value contract.
- Wired through the runtime Drizzle builder (`pgNumeric`/`mysqlDecimal`/`sqliteNumeric`) so runtime tables and the diff snapshot agree.
- Corrected the JSDoc examples that promised decimals by default.

## Why it doesn't churn the diff

`normalizeType` strips precision (`numeric(10,2)` and `decimal(10,2)` both collapse to `numeric`), so the descriptor's `dialectType` matches what introspection returns on every dialect. A regression test asserts this round-trip.

## Design decisions (confirmed with the maintainer)

- **Opt-in, integer default** (non-breaking) rather than changing the default to decimal.
- **Exact `DECIMAL`/`NUMERIC`** rather than `double` (approximate, unsuitable for money).
- The builder's existing `options.format === "float"` (→ `double`) mapping is left unchanged; converging it onto exact decimal is a noted follow-up.

## Scope note

This fix covers the collection code-first path (the descriptor + runtime generator, the audit's documented surface). The separate legacy component/user-field codegen (`component-schema-service`, `user-ext-schema-service`) still emits `double` for number fields; converging those is a follow-up.

## Verification

- New tests: 5 descriptor cases (all dialects + the no-phantom-diff `normalizeType` round-trip) and 2 runtime-generator cases (asserting `integer` default and `numeric(10, 2)` via `getSQLType()`). All pass.
- `pnpm --filter nextly check-types` + `lint` clean; `pnpm build` green (15/15).
- Pre-existing baseline: the schema-services/diff suites carry 6 pre-existing failures on `main` (stale `number → double/float8/real` expectations from the descriptor refactor + a layout-skip test); this PR adds **0** new failures and does not touch them.

## Summary by CodeRabbit

* **New Features**
  * Number fields default to whole-number storage.
  * Added opt-in support for exact decimal values using configurable precision and scale.
  * Decimal storage works consistently across PostgreSQL, MySQL, and SQLite; float formatting remains supported.
* **Documentation**
  * Updated number-field guidance and examples to clarify integer vs decimal storage options.
* **Bug Fixes**
  * Migration drift detection and migration generation now preserve decimal storage details correctly.
* **Validation**
  * Improved config validation for decimal precision/scale, including clearer error cases.
* **Tests**
  * Added coverage for decimal mapping and SQL type generation.

### CODE REVIEW

File: packages/nextly/src/domains/schema/services/runtime-schema-generator.ts [hunk 1]
```diff
@@ -234,23 +239,41 @@ function buildUserDrizzleColumn(
...
+  scale: number;
+  mode: "number";
+} {
+  // `mode: "number"` reads the column back as a JS number, matching the number
+  // field's value contract (the same contract the `double` kind honors).
+  return {
+    precision: desc.precision ?? 10,
+    scale: desc.scale ?? 2,
+    mode: "number",
+  };
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/runtime-schema-generator.ts:271]: Fixed in 40472d64. `DEFAULT_DECIMAL_PRECISION`/`DEFAULT_DECIMAL_SCALE` are now exported from `field-column-descriptor.ts` and imported by `decimalConfig`, so the fallbacks reference the shared source and the two files cannot drift.

File: packages/nextly/src/collections/fields/types/number.ts [hunk 2]
```diff
@@ -165,6 +168,41 @@ export interface NumberFieldConfig
...
+   * Number of digits to the right of the decimal point for a
+   * `dbType: "decimal"` column. Ignored for integer fields.
+   *
+   * @default 2
+   */
+  scale?: number;
+
   /**
    * Allow multiple numeric values.
    *
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/collections/fields/types/number.ts:212]: Fixed in 40472d64. Added `validateNumberDecimalDimensions` in `validate-config.ts`: it rejects `scale > precision`, `precision < 1`, non-integers, and negative scale at config time (with tests), so the bad size surfaces as a config error instead of an opaque DDL failure.

File: packages/nextly/src/collections/fields/types/number.ts [hunk 3]
```diff
@@ -78,13 +79,15 @@ export interface NumberFieldAdminOptions extends FieldAdminOptions {
...
- * // Basic number field
+ * // Price field: exact decimal storage (integer is the default and truncates)
  * const priceField: NumberFieldConfig = {
  *   name: 'price',
  *   type: 'number',
  *   label: 'Price',
  *   required: true,
  *   min: 0,
+ *   dbType: 'decimal',
+ *   scale: 2,
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/collections/fields/types/number.ts:90]: Fixed in 40472d64. The JSDoc no longer says integer always "truncates"; it now states a fractional value is not preserved (the database rounds or truncates it), and points to `dbType: "decimal"` when fractions must be kept.

File: packages/nextly/src/collections/fields/types/number.ts [hunk 4]
```diff
@@ -165,6 +168,41 @@ export interface NumberFieldConfig
...
+   */
+  precision?: number;
+
+  /**
+   * Number of digits to the right of the decimal point for a
+   * `dbType: "decimal"` column. Ignored for integer fields.
+   *
+   * @default 2
+   */
+  scale?: number;
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/collections/fields/types/number.ts:208]: Fixed in 40472d64. `precision`/`scale` are validated at config time (positive-integer precision, non-negative-integer scale, `scale <= precision`), rejecting zero/negative/fractional/NaN before any DDL is rendered. Tests in `validate-config.decimal.test.ts`.

File: packages/nextly/src/domains/schema/services/field-column-descriptor.ts [hunk 5]
```diff
@@ -158,9 +184,13 @@ function classifyFieldKind(field: FieldDefinition): ColumnKind {
       return "longText";
 
     case "number": {
-      // UI-created fields carry options.format === "float" for decimal storage;
-      // code-first fields have no options, defaulting to integer — matching
-      // the DDL emitted by dynamic-collection-schema-service.ts.
+      // Code-first fields opt into exact fractional storage via
+      // `dbType: "decimal"` (DECIMAL/NUMERIC), the right choice for money.
+      const dbType = (field as { dbType?: string }).dbType;
+      if (dbType === "decimal") return "decimal";
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.ts]: Fixed in 40472d64. `dbType`/`precision`/`scale` are now first-class on the runtime `FieldDefinition`, so `getColumnDescriptor` reads them typed (no cast) and the tests use typed fixtures.

File: packages/nextly/src/domains/schema/__tests__/runtime-schema-generator.test.ts [hunk 6]
```diff
@@ -91,6 +91,23 @@ describe("generateRuntimeSchema", () => {
...
+      const fields: FieldDefinition[] = [{ name: "count", type: "number" }];
+      const result = generateRuntimeSchema("dc_products", fields, "postgresql");
+      const col = getColumns(result.table).count as { getSQLType(): string };
+      expect(col.getSQLType()).toBe("integer");
+    });
+
+    it("maps a dbType:'decimal' number field to an exact numeric column", () => {
+      const fields: FieldDefinition[] = [
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        { name: "price", type: "number", dbType: "decimal", scale: 2 } as any,
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/__tests__/runtime-schema-generator.test.ts]: Fixed in 40472d64. `FieldDefinition` now carries `dbType`/`precision`/`scale`, so the decimal test uses a fully typed field with no `as any` / eslint-disable.

File: packages/nextly/src/domains/schema/services/field-column-descriptor.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.ts]: Fixed in 40472d64. Both `migrate:create` and `migrate:check` field adapters now forward `dbType`/`precision`/`scale`, so CLI-generated migrations and drift checks see the decimal field as decimal rather than the integer default.

File: packages/nextly/src/domains/schema/services/field-column-descriptor.ts [hunk 7]
```diff
@@ -124,13 +133,30 @@ export function getColumnDescriptor(
...
-  const dialectType = renderDialectType(kind, dialect, /* length */ undefined);
+  // Decimal fields carry precision/scale (author-set or the DECIMAL(10,2)
+  // default); every other kind ignores them.
+  const decimal =
+    kind === "decimal"
+      ? {
+          precision:
+            (field as { precision?: number }).precision ??
+            DEFAULT_DECIMAL_PRECISION,
+          scale: (field as { scale?: number }).scale ?? DEFAULT_DECIMAL_SCALE,
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.ts]: Fixed in 40472d64. `validateNumberDecimalDimensions` rejects invalid `precision`/`scale` (non-integer, negative, `scale > precision`) at config time, before the value reaches DDL.

File: packages/nextly/src/domains/schema/services/runtime-schema-generator.ts [hunk 8]
```diff
@@ -325,6 +357,12 @@ function buildSqliteColumnFromKind(
       return nullable ? sqliteInteger(name) : sqliteInteger(name).notNull();
     case "double":
       return nullable ? sqliteReal(name) : sqliteReal(name).notNull();
+    case "decimal": {
+      // SQLite has no fixed-precision decimal; NUMERIC affinity is the closest,
+      // read back as a JS number to match the field's value contract.
+      const col = sqliteNumeric(name, { mode: "number" });
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/runtime-schema-generator.ts:365]: Fixed in 40472d64. The JSDoc and changeset now state SQLite maps to NUMERIC affinity (best-effort, no fixed-precision decimal type); exact-at-rest is claimed only for Postgres and MySQL.

File: packages/nextly/src/domains/schema/services/runtime-schema-generator.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/runtime-schema-generator.ts:271]: Addressed in 40472d64 (documented rather than changed). Storage is exact on Postgres/MySQL; the value is read back as a JS number to honor the number field's value contract, which is lossless for a decimal(10,2) (well within double precision). The JSDoc now states this and that a precision beyond ~15 significant digits is not fully round-tripped; SQLite is documented as best-effort. Switching to `mode: "string"` would make the field value a string and break the number-typed contract, so the JS-number read is the intended trade-off.

File: packages/nextly/src/domains/schema/services/field-column-descriptor.test.ts [hunk 9]
```diff
@@ -35,3 +45,65 @@ describe("getColumnDescriptor — plugin field types", () => {
...
+      const d = getColumnDescriptor(
+        numberField({ dbType: "decimal" }),
+        dialect
+      );
+      expect(normalizeType(d?.dialectType)).toBe("numeric");
+    }
+    // Introspection-shaped inputs normalize to the same token.
+    expect(normalizeType("numeric")).toBe("numeric");
+    expect(normalizeType("decimal(10,2)")).toBe("numeric");
+  });
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.test.ts:111]: Verified and real: `normalizeType` strips `(p,s)`, so a decimal resize (e.g. 10,2 -> 12,4) on an existing column is not emitted as a diff. Detecting it safely means teaching the live introspector to read `numeric_precision`/`numeric_scale` on all three dialects and comparing them only when both sides carry precision, plus golden coverage so no *other* numeric column starts churning. That is a separate, careful change to the diff/introspection core. Since decimals are brand new (no existing decimal columns to resize yet) and the current behavior is a known-limitation, not data loss, I documented it explicitly in `renderDialectType` and am leaving this thread open for you to decide whether to schedule the introspection follow-up now. Not resolving, since this is a deliberate deferral rather than a fix.

File: packages/nextly/src/domains/schema/services/field-column-descriptor.ts [hunk 10]
```diff
@@ -211,8 +241,18 @@ function classifyFieldKind(field: FieldDefinition): ColumnKind {
...
+  opts: { length?: number; precision?: number; scale?: number }
 ): string {
+  const { length } = opts;
+  if (kind === "decimal") {
+    // The diff normalizes precision away (numeric(10,2) -> "numeric"), so these
+    // never cause a phantom type change; precision is carried for the emitter.
+    const p = opts.precision ?? DEFAULT_DECIMAL_PRECISION;
+    const s = opts.scale ?? DEFAULT_DECIMAL_SCALE;
+    if (dialect === "postgresql") return `numeric(${p}, ${s})`;
+    if (dialect === "mysql") return `decimal(${p},${s})`;
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.ts:258]: Verified and real: `normalizeType` strips `(p,s)`, so a decimal resize (e.g. 10,2 -> 12,4) on an existing column is not emitted as a diff. Detecting it safely means teaching the live introspector to read `numeric_precision`/`numeric_scale` on all three dialects and comparing them only when both sides carry precision, plus golden coverage so no *other* numeric column starts churning. That is a separate, careful change to the diff/introspection core. Since decimals are brand new (no existing decimal columns to resize yet) and the current behavior is a known-limitation, not data loss, I documented it explicitly in `renderDialectType` and am leaving this thread open for you to decide whether to schedule the introspection follow-up now. Not resolving, since this is a deliberate deferral rather than a fix.

File: packages/nextly/src/collections/config/validate-config.ts [hunk 11]
```diff
@@ -403,6 +406,60 @@ function validateField(
...
+  if (
+    Number.isInteger(effPrecision) &&
+    Number.isInteger(effScale) &&
+    effScale > effPrecision
+  ) {
+    errors.push({
+      path: `${path}.scale`,
+      message: "Decimal 'scale' cannot exceed 'precision'",
+      code: "DECIMAL_SCALE_EXCEEDS_PRECISION",
+    });
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/collections/config/validate-config.ts]: Fixed in 26a5acbd. Decimal `precision`/`scale` are now bounded to the most restrictive supported-dialect maximum (MySQL: precision 1..65, scale 0..30) so the config stays portable, with the boundary cases covered in tests (66 and scale 31 rejected, 65 accepted). Chose portable bounds over dialect-aware validation since a decimal field should apply cleanly on any of the three databases.

File: packages/nextly/src/collections/config/validate-config.ts [hunk 12]
```diff
@@ -403,6 +406,60 @@ function validateField(
...
+ * at the database with an opaque DDL error on first apply, so reject them at
+ * config time: precision a positive integer, scale a non-negative integer no
+ * larger than precision.
+ */
+function validateNumberDecimalDimensions(
+  f: Record<string, unknown>,
+  path: string,
+  errors: ValidationError[]
+): void {
+  if (f.dbType !== "decimal") return;
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/collections/config/validate-config.ts]: Fixed in 75be8ea3. The decimal-dimensions check moved to a shared validator (`validateNumberDecimalDimensionsShared` in base-validator.ts) and is now called from the collection, single, AND component field validators, so an invalid `precision`/`scale` is rejected on every surface. Direct unit test added in base-validator.decimal.test.ts.

File: packages/nextly/src/domains/schema/services/field-column-descriptor.ts [hunk 13]
```diff
@@ -158,11 +182,13 @@ function classifyFieldKind(field: FieldDefinition): ColumnKind {
...
 
     case "number": {
-      // UI-created fields carry options.format === "float" for decimal storage;
-      // code-first fields have no options, defaulting to integer — matching
-      // the DDL emitted by dynamic-collection-schema-service.ts.
-      const fmt = (field as { options?: { format?: string } }).options?.format;
-      return fmt === "float" ? "double" : "integer";
+      // Code-first fields opt into exact fractional storage via
+      // `dbType: "decimal"` (DECIMAL/NUMERIC), the right choice for money.
+      if (field.dbType === "decimal") return "decimal";
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.ts:191]: Confirmed and valid. Adding `dbType` to the shared field contract made `dbType: "decimal"` *accepted* on singles/components/dynamic collections, but those surfaces build their tables through the legacy `DynamicCollectionSchemaService.generateMigrationSQL` / `ComponentSchemaService.getColumnType`, which key off `options.format` and ignore `dbType`, so the column is still integer/real there. Storage is honored only on the modern descriptor path (code-first collections + the diff/runtime table), which is the surface this PR targeted (audit finding P0-G). Config-time validation of the dimensions now runs on all three surfaces (75be8ea3), so a bad value is rejected everywhere; what remains is emitting a decimal column in the legacy generators. That is the same "legacy DDL path is behind the descriptor" class as several other audit F-findings, and wiring precision/scale through those two generators is a further change to the legacy DDL layer. I am raising it with the maintainer to decide whether to fold it into this PR or track it as the immediate follow-up, and will not leave it silently ignored. Keeping this thread open pending that decision.

File: packages/nextly/src/collections/fields/types/number.ts [hunk 14]
```diff
@@ -165,6 +168,45 @@ export interface NumberFieldConfig
...
+   *   round-tripped.
+   *
+   * @default "integer"
+   * @example
+   * ```typescript
+   * // Store a price, e.g. 1234.56
+   * { name: 'price', type: 'number', dbType: 'decimal', precision: 10, scale: 2 }
+   * ```
+   */
+  dbType?: "integer" | "decimal";
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/collections/fields/types/number.ts:192]: Confirmed and valid. Adding `dbType` to the shared field contract made `dbType: "decimal"` *accepted* on singles/components/dynamic collections, but those surfaces build their tables through the legacy `DynamicCollectionSchemaService.generateMigrationSQL` / `ComponentSchemaService.getColumnType`, which key off `options.format` and ignore `dbType`, so the column is still integer/real there. Storage is honored only on the modern descriptor path (code-first collections + the diff/runtime table), which is the surface this PR targeted (audit finding P0-G). Config-time validation of the dimensions now runs on all three surfaces (75be8ea3), so a bad value is rejected everywhere; what remains is emitting a decimal column in the legacy generators. That is the same "legacy DDL path is behind the descriptor" class as several other audit F-findings, and wiring precision/scale through those two generators is a further change to the legacy DDL layer. I am raising it with the maintainer to decide whether to fold it into this PR or track it as the immediate follow-up, and will not leave it silently ignored. Keeping this thread open pending that decision.

File: packages/nextly/src/schemas/dynamic-collections/legacy-types.ts [hunk 15]
```diff
@@ -69,6 +69,14 @@ export type FieldDefinition = {
   default?: unknown;
   length?: number;
 
+  /** Storage type for number fields: whole-number `integer` (default) or exact
+   * fixed-point `decimal` (sized by `precision`/`scale`). */
+  dbType?: "integer" | "decimal";
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/schemas/dynamic-collections/legacy-types.ts:74]: Confirmed and valid. Adding `dbType` to the shared field contract made `dbType: "decimal"` *accepted* on singles/components/dynamic collections, but those surfaces build their tables through the legacy `DynamicCollectionSchemaService.generateMigrationSQL` / `ComponentSchemaService.getColumnType`, which key off `options.format` and ignore `dbType`, so the column is still integer/real there. Storage is honored only on the modern descriptor path (code-first collections + the diff/runtime table), which is the surface this PR targeted (audit finding P0-G). Config-time validation of the dimensions now runs on all three surfaces (75be8ea3), so a bad value is rejected everywhere; what remains is emitting a decimal column in the legacy generators. That is the same "legacy DDL path is behind the descriptor" class as several other audit F-findings, and wiring precision/scale through those two generators is a further change to the legacy DDL layer. I am raising it with the maintainer to decide whether to fold it into this PR or track it as the immediate follow-up, and will not leave it silently ignored. Keeping this thread open pending that decision.

File: packages/nextly/src/singles/config/validate-single.ts [hunk 16]
```diff
@@ -220,6 +221,10 @@ function validateField(
     case "component":
       validateComponentFieldRefShared(f, path, errsBase);
       break;
+
+    case "number":
+      validateNumberDecimalDimensionsShared(f, path, errsBase);
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/singles/config/validate-single.ts:229]: Fixed in 9cb9930c. Added DECIMAL_PRECISION_INVALID / DECIMAL_SCALE_INVALID / DECIMAL_SCALE_EXCEEDS_PRECISION to both SingleValidationErrorCode and ComponentValidationErrorCode.

File: packages/nextly/src/domains/schema/services/field-column-descriptor.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.ts:191]: Fixed in 9cb9930c. A `hasMany` number now maps to a JSON column (hasMany wins over dbType), so the stringified array is stored in JSON not a scalar numeric column. Test added.

[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.ts:191]: Maintainer-approved follow-up. Adding `dbType` to the shared contract made decimal accepted on this surface, but it builds tables through the legacy DDL generator, which emits integer/real. Config-time validation already runs here (75be8ea3) so bad values are rejected; emitting a decimal column in those generators is tracked in tasks/drizzle-v1-findings-verification.md so PR #216 (code-first collection decimals) stays reviewable. Resolving as a tracked follow-up, not a fix here.

File: packages/nextly/src/collections/fields/types/number.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/collections/fields/types/number.ts:192]: Maintainer-approved follow-up. Adding `dbType` to the shared contract made decimal accepted on this surface, but it builds tables through the legacy DDL generator, which emits integer/real. Config-time validation already runs here (75be8ea3) so bad values are rejected; emitting a decimal column in those generators is tracked in tasks/drizzle-v1-findings-verification.md so PR #216 (code-first collection decimals) stays reviewable. Resolving as a tracked follow-up, not a fix here.

File: packages/nextly/src/schemas/dynamic-collections/legacy-types.ts (hunk 15 above)
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/schemas/dynamic-collections/legacy-types.ts:74]: Maintainer-approved follow-up. Adding `dbType` to the shared contract made decimal accepted on this surface, but it builds tables through the legacy DDL generator, which emits integer/real. Config-time validation already runs here (75be8ea3) so bad values are rejected; emitting a decimal column in those generators is tracked in tasks/drizzle-v1-findings-verification.md so PR #216 (code-first collection decimals) stays reviewable. Resolving as a tracked follow-up, not a fix here.

File: packages/nextly/src/collections/fields/types/number.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/collections/fields/types/number.ts:192]: Maintainer-approved follow-up. Adding `dbType` to the shared contract made decimal accepted on this surface, but it builds tables through the legacy DDL generator, which emits integer/real. Config-time validation already runs here (75be8ea3) so bad values are rejected; emitting a decimal column in those generators is tracked in tasks/drizzle-v1-findings-verification.md so PR #216 (code-first collection decimals) stays reviewable. Resolving as a tracked follow-up, not a fix here.

File: packages/nextly/src/domains/schema/services/field-column-descriptor.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/domains/schema/services/field-column-descriptor.ts:258]: Maintainer-approved follow-up. Detecting a decimal resize needs the live introspector to read numeric_precision/numeric_scale across dialects plus golden coverage; tracked in tasks/drizzle-v1-findings-verification.md. Resolving as a tracked follow-up.

File: packages/nextly/src/schemas/dynamic-collections/legacy-types.ts [hunk 17]
```diff
@@ -69,6 +69,14 @@ export type FieldDefinition = {
...
+  /** Total significant digits for a `decimal` number field (default 10). */
+  precision?: number;
+  /** Fractional digits for a `decimal` number field (default 2). */
+  scale?: number;
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/schemas/dynamic-collections/legacy-types.ts:78]: Maintainer-approved follow-up. This is the Builder/ui-schema surface: `uiSchemaFieldSchema` + `uiToMinimal` drop `dbType`/`precision`/`scale`, so a UI-created decimal field loses the settings. Same "builder/dynamic path does not carry decimal" class as the legacy-DDL follow-up, tracked in tasks/drizzle-v1-findings-verification.md. PR #216 delivers code-first collection decimals; the builder/dynamic path is being evaluated (modernize vs remove) in a researched follow-up. Resolving as tracked follow-up.

[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/schemas/dynamic-collections/legacy-types.ts:78]: Maintainer-approved follow-up. The dynamic/API validation path (`DynamicCollectionValidationService`) does not yet call the decimal-dimension validator, so a Builder/API-authored decimal field is not dimension-checked (the code-first collection/single/component validators are). Same builder/dynamic surface as above; tracked in tasks/drizzle-v1-findings-verification.md for the researched follow-up. Resolving as tracked follow-up.
